src/services/web3/connectors/connector.ts

```diff
@@ -0,0 +1,69 @@
+import { ref, Ref } from 'vue';
+import Web3 from 'web3';
+import { getAddress } from '@ethersproject/address';
+
+export type ConnectorPayload = {
+  provider: Web3;
+  account: Ref<string | null>;
+  chainId: Ref<number | null>;
+};
+export abstract class Connector {
+  provider: any = null;
+  account: Ref<string | null> = ref(null);
+  chainId: Ref<number | null> = ref(null);
+  active: Ref<boolean> = ref(false);
+  selectedAccount = '';
+  id = 'unknown';
```

Comment:
I think we should add a `signer` here as well so that it's easily available when you need it. (its essentially calling `getSigner()` on the provider)

Reply:
I believe most provider consumers will extract the signer for you if you simply provide a signer capable provider? I could be wrong

Reply:
That's an awesome idea, I wish it worked this way - but it seems you have to grab the signer from the provider to do wallet stuff (see `sendTransaction` function in `web3.ts`)

Reply:
added in a signer which gets returned from the composable :)